homeassistant/components/openweathermap/weather.py

```diff
@@ -203,18 +203,16 @@ def calc_precipitation(rain, snow):
                     }
                 )
             else:
+                rain = entry.get_rain().get("1h")
```

Comment:
Of `entry.get_rain` might return `None` we need to check for that before doing the dict lookup.

Reply:
As of the current API, this should only be the case for 'daily' which is handled above. This was also not part of the logic before. But I can add this extra check, if you want.

Reply:
No, if it can't happen we don't need to guard it.